wpilibc/src/main/native/cpp/simulation/Mechanism2D.cpp

```diff
@@ -0,0 +1,44 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "frc/simulation/Mechanism2D.h"
+
+#include <wpi/SmallString.h>
+#include <wpi/Twine.h>
+#include <wpi/raw_ostream.h>
+
+using namespace frc;
+
+Mechanism2D::Mechanism2D() : m_device{"Mechanism2D"} {}
+
+void Mechanism2D::SetLigamentAngle(const wpi::Twine& ligamentPath,
+                                   float angle) {
+  if (m_device) {
+    wpi::SmallString<64> fullPathBuf;
+    wpi::StringRef fullPath =
+        (ligamentPath + "angle/").toNullTerminatedStringRef(fullPathBuf);
```

Comment:
Should we require the user put a trailing / on the ligamentPath?  Or should this be appending "/angle/"?

Reply:
That's a good point. /angle/ does make more sense.

Reply:
I ended up changing it to jasonOne/ligamentOne/ligamentTwo instead of /jasonOne/ligamentOne/ligamentTwo/ to save the two characters.